pom.xml

```diff
@@ -79,8 +79,8 @@
 
    <properties>
       <maven.compiler.source>11</maven.compiler.source>
-      <maven.compiler.target>11</maven.compiler.target>
-      <maven.compiler.release>11</maven.compiler.release>
+      <maven.compiler.target>17</maven.compiler.target>
+      <maven.compiler.release>17</maven.compiler.release>
```

Comment:
Why not changing the source level too?

Reply:
I left the source at 11 to facilitate back-porting fixes if necessary. Java 17 only has 2 language-level changes (i.e. sealed classes & pattern matching for switch expressions and statements) which I don't think we'll adopt anytime soon, but I think it makes sense to have a formal guard in place just to be safe. We can move the source to 17 in 3.0 (assuming we drop support for 2.x at that point). This is the same approach as taken [by Classic](https://github.com/apache/activemq/blob/52a9aec22ec8309312cdd1a223f2f61dfbc83b2f/pom.xml#L117-118).

Reply:
Ok to hold off to 3.0.0 I guess, though in general I'd just be updating it at the same time as the others.

Dont forget the other features from 12-16 such as Records etc. (BTW 'Pattern matching for switch' in 17 was a preview; there was 'expressions in switch' from 14, or 'patterns in instanceof' from 16).

I think Classic is mainly doing that because changing it causes compile failures in the existing source. Since we are using maven.compiler.release 17 in this case (meaning it compiles with the 17 signatures regardless), which classic is not, I'm not sure there is the same scope for compile failure here.

Reply:
For posterity's sake, I changed my mind on this. I'm pushing the source to 17 as well.